File: Codes/3349-adjacent-increasing-subarrays-detection-i.go
/*
 * @Author: LetMeFly
 * @Date: 2025-10-14 20:14:45
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-10-14 20:26:58
 */
package main


func isOk3349(nums []int, i, k int) bool {
    for j := 0; j + 1 < k; j++ {
        if nums[i + j] >= nums[i + j + 1] {
            return false
        }
    }
    return true
}

func hasIncreasingSubarrays(nums []int, k int) bool {
    for i := 0; i + 2 * k <= len(nums); i++ {
        if isOk3349(nums, i, k) && isOk3349(nums, i + k, k) {
            return true
        }
    }
    return false
}

File: Codes/3349-adjacent-increasing-subarrays-detection-i.java
/*
 * @Author: LetMeFly
 * @Date: 2025-10-14 20:14:45
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-10-14 20:30:13
 */
// THIS IS NOT RIGHT
class Solution {
    private boolean isOk(List<Integer> nums, int i, int k) {
        while (--k > 0) {
            if (nums[i] >= nums[++i]) {
                return false;
            }
        }
        return true;
    }

    public boolean hasIncreasingSubarrays(List<Integer> nums, int k) {
        for (int i = 0; i + 2 * k <= nums.length(); i++) {
            if (isOk(nums, i, k) && isOk(nums, i + k, k)) {
                return true;
            }
        }
        return false;
    }
}